### PR TITLE
DS-2218: Unable to use command "update-handle-prefix"

### PR DESCRIPTION
**Pull request by ctu-developers**
https://jira.duraspace.org/browse/DS-2218

Use the command "bin/dspace update-handle-prefix <old handle> <new handle>" ends with the "ORA-00911: invalid character". The problem is the extra semicolon in sql command.

I made changes in formatting SQL.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/handle/UpdateHandlePrefix.java [hunk 1]
```diff
@@ -9,89 +9,159 @@
...
+                        );
 
-                System.out.print("Re-creating browse and search indexes... ");                
+                        System.out.print("Updating metadatavalues table... ");
+                        sql = "UPDATE metadatavalue " +
+                              "SET text_value = " +
+                                "(" +
+                                  "SELECT 'http://hdl.handle.net/' || handle " +
+                                  "FROM handle " +
+                                  "WHERE handle.resource_id = item_id AND handle.resource_type_id = 2" +
```
Comment (helix84) [inline dspace-api/src/main/java/org/dspace/handle/UpdateHandlePrefix.java]: It blows up on this line:

Updating handle table... 2535 items updated
Updating metadatavalues table... 
Error during SQL operations.
Exception: ERROR: column "item_id" does not exist
  Position: 121
org.postgresql.util.PSQLException: ERROR: column "item_id" does not exist
  Position: 121
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2103)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1836)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:257)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:512)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(AbstractJdbc2Statement.java:388)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.executeUpdate(AbstractJdbc2Statement.java:334)
        at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
        at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
        at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
        at org.dspace.storage.rdbms.DatabaseManager.updateQuery(DatabaseManager.java:441)
        at org.dspace.handle.UpdateHandlePrefix.main(UpdateHandlePrefix.java:103)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.dspace.app.launcher.ScriptLauncher.runOneCommand(ScriptLauncher.java:225)
        at org.dspace.app.launcher.ScriptLauncher.main(ScriptLauncher.java:77)

**Comment by ctu-developers** [comment]
Hmm, original code was for 4.x version:-(

File: dspace-api/src/main/java/org/dspace/handle/UpdateHandlePrefix.java (hunk 1 above)
Comment (ctu-developers) [inline dspace-api/src/main/java/org/dspace/handle/UpdateHandlePrefix.java]: Hmm, original code was for 4.x version:-(

**Comment by ctu-developers** [comment]
The latest commit repairing update of metadata table. First prevent to update handle to NULL value, and then ensures the updating only data set corresponding to the original handle, not completely full metadata handle values.

Mergeable to *dspace-4_x* branch only (backport)! For *master/dspace-5_x* see #929.

**Comment by helix84** [comment]
Wrong target branch, rebased on dspace-4_x in #930.

**Pull request closed without merging**